symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedLoopV2.java

```diff
@@ -45,13 +47,24 @@
 @API(status = API.Status.INTERNAL)
 public class DatafeedLoopV2 extends AbstractDatafeedLoop {
 
+  /**
+   * DFv2 API authorizes a maximum length for the tag parameter,
+   * this might be an issue if bots are using very long names,
+   * in that case the differentiating part between bot's names
+   * should be put first.
+   */
+  private static final int DATAFEED_TAG_MAX_LENGTH = 100;
+
   private final AtomicBoolean started = new AtomicBoolean();
   private final AckId ackId;
+  private final String tag;
+
   private V5Datafeed datafeed;
 
   public DatafeedLoopV2(DatafeedApi datafeedApi, AuthSession authSession, BdkConfig config) {
     super(datafeedApi, authSession, config);
     this.ackId = new AckId().ackId("");
+    this.tag = StringUtils.truncate(bdkConfig.getBot().getUsername(), DATAFEED_TAG_MAX_LENGTH);
```

Comment:
Should we log a warning if the bot username is greater than ` DATAFEED_TAG_MAX_LENGTH` ?

Reply:
We could but at the same time it could be a valid use case if you made sure that your bot's name starts with a differentiating part (bot1_longstring... and bot2_longstring...)

What do you think @symphony-thibault ?

Reply:
As `bdkConfig.getBot().getUsername()` is the value used for authentication, I don't really see how we could end up with `bot1_longstring...` and `bot2_longstring...`

Reply:
Something else to consider, the tag is specific for each user. So it is very unlikely that truncating the bot's service account name to <100 characters is going to be a problem as most of the time a service account is used for a single bot.